src/Game/Scripting/Commands.cs

```diff
@@ -24,845 +24,1062 @@
-            Interpreter.RegisterCommandHandler("clearbuy", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("clearsell", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("organizer", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("counter", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("replygump", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("closegump", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("cleartargetqueue", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargetlast", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargetself", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargetobject", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargettype", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargettile", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargettileoffset", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargettilerelative", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargetghost", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("autotargetground", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("cancelautotarget", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("getenemy", UnimplementedCommand);
-            Interpreter.RegisterCommandHandler("getfriend", UnimplementedCommand);
+            DateTime lastExecuted = DateTime.UtcNow.AddHours(-2.0); // 2 hours ago
+            foreach(string key in commandKeywords)
+            {
+                if (_lastExecutedRegistry.ContainsKey(key) && _lastExecutedRegistry[key] > lastExecuted)
+                    lastExecuted = _lastExecutedRegistry[key];
+            }
+            return lastExecuted;
         }
 
-        private static bool UnimplementedCommand(string command, Argument[] args, bool quiet, bool force)
+        // Build execution
+        public CommandExecution(Command command, ParameterList parameters, bool force)
         {
-            GameActions.Print($"Unimplemented command: '{command}'", type: MessageType.System);
-            return true;
+            Cmd = command;
+            Params = parameters;
+            Force = force;
+            CreationTime = DateTime.UtcNow;
         }
 
-        private static bool Deprecated(string command, Argument[] args, bool quiet, bool force)
+        // Inspired by Razor souce code, this method check if it is time to execute the command
+        // and if the time is right, perform the execution
+        // Logic inside may be complex, checking for a given target or waiting for a given gump
+        public bool PerformWait()
         {
```

Comment:
I am SOOOO HAPPY to read this!
I fully agree, but as this was how I found implemented in Razor (for example in the WalkAction class), I decided to follow it as-is imagining it could be a method engineers are familiar with.
But now that I know this is also confusing for you I will change it for sure \o/